mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,247 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import ast
+import json
+import os
+import re
+from botocore.vendored import requests
+import logging
+import secret_manager
+
+
+class LabelBot:
+    LABEL_PAGE_PARSE = 30  # Limit for total labels per page to parse
+
+    def __init__(self,
+                 repo=os.environ.get("repo"),
+                 github_user=None,
+                 github_oauth_token=None,
+                 apply_secret=True):
+        """
+        Initializes the Label Bot
+        :param repo: GitHub repository that is being referenced
+        :param github_user: GitHub username
+        :param github_oauth_token: GitHub authentication token (Personal access token)
+        :param apply_secret: GitHub secret credential (Secret credential that is unique to a GitHub developer)
+        """
+        self.repo = repo
+        self.github_user = github_user
+        self.github_oauth_token = github_oauth_token
+        if apply_secret:
+            self.get_secret()
+        self.auth = (self.github_user, self.github_oauth_token)
+        self.all_labels = None
+
+    def get_rate_limit(self):
+        """
+        This method gets the remaining rate limit that is left from the GitHub API
+        :return Remaining API requests left that GitHub will allow
+        """
+        res = requests.get('https://api.github.com/{}'.format('rate_limit'),
+                           auth=self.auth)
+        res.raise_for_status()
+        data = res.json()['rate']
+        return data['remaining']
+
+    def get_secret(self):
+        """
+        This method is to get secret value from Secrets Manager
+        """
+        secret = json.loads(secret_manager.get_secret())
+        self.github_user = secret["github_user"]
+        self.github_oauth_token = secret["github_oauth_token"]
+
+    def tokenize(self, string):
+        """
+        This method is to extract labels from comments
+        :param string: String parsed from a GitHub comment
+        :return Set of Labels which have been extracted
+        """
+        substring = string[string.find('[') + 1: string.rfind(']')]
+        labels = [' '.join(label.split()) for label in substring.split(',')]
+        return labels
+
+    def clean_string(self, raw_string, sub_string):
+        """
+        This method is to convert all non-alphanumeric characters from raw_string to sub_string
+        :param raw_string The original string messy string
+        :param sub_string The string we want to convert to
+        :return Fully converted string
+        """
+        cleans = re.sub("[^0-9a-zA-Z]", sub_string, raw_string)
```

Comment:
move regex into a constant describing what it does - for example this could be ALPHANUMERIC = [^0-9a-zA-Z]
Rename `cleans` to be more understandable - how about `converted_string` or `response`?